### PR TITLE
MNTOR-978: Migrate breach resolution (part 3)

### PR DESCRIPTION
**Pull request by mansaj**
# References: 
Jira: MNTOR-978
Figma: n/a

# Description
After #2766 is merged and ran, the `breach_resolution` column should be updated to now use breachId as the key for resolutions

This PR modifies the way breach resolutions are retrieved to conform to the new key. Instead of using recency Index, we now have a flag to determine whether the breach resolution json has been migrated or not. If it has, we retrieve the json object and parse it based on breachId. If not, we fall back to using recency index. The GET is meant to be backwards compatible.

This PR also modifies the way breach resolutions are saved to the database by replacing `recencyIndex` with `breachId` 

# Screenshot (if applicable)

Not applicable.

# How to test
- After running `deprecateBreachResolve` and `recencyToBreachId` on your local db, try loading breaches page to test regression
- Try making a PUT call via `api/v1/breaches` and make sure the call returns a `200`

# Checklist (Definition of Done)
- [x] Localization strings (if needed) have been added.
- [x] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).
- [x] I've added or updated the relevant sections in readme and/or code comments
- [x] I've added a unit test to test for potential regressions of this bug.
- [x] Product Owner accepted the User Story (demo of functionality completed) or waived the privilege.
- [x] All acceptance criteria are met.
- [x] Jira ticket has been updated (if needed) to match changes made during the development process.
- [x] Jira ticket has been updated (if needed) with suggestions for QA when this PR is deployed to stage.

### CODE REVIEW

**Review by toufali — APPROVED** [review]
Looks like you're merging this branch into `MNTOR-978` – do you plan to do that first before merging `MNTOR-978` into `main`?  No issue, just checking.

After running this branch, I can confirm a breach that I had set as "resolved" was put back into the correct column.  (Without this branch, it was incorrectly listed as "unresolved").

I got a 404 when attempting to PUT `/api/v1/breaches`.  It's possible I ran my curl command incorrectly...

**Comment by mansaj** [comment]
So the part 2 PR has to go in first. We will execute it to prepare the db,
then we can merge in this branch.

I set the destination branch as mntor-978 only because its easier/clearer
to review.. I just wanted to show the actual changes for this part. If I
use main then it will show both the changes for part 2 and part 3.

Should the put be /api/v1/user/breaches?

File: src/views/partials/breaches.js [hunk 1]
```diff
@@ -63,7 +63,7 @@ function createResolveSteps (breach) {
   const checkedArr = breach.ResolutionsChecked || []
   const resolveStepsHTML = Object.entries(breach.breachChecklist).map(([key, value]) => `
   <li class='resolve-list-item'>
-    <input name='${breach.recencyIndex}' value='${key}' type='checkbox' ${checkedArr.includes(key) ? 'checked' : ''}>
+    <input name='${breach.Id}' value='${key}' type='checkbox' ${checkedArr.includes(key) ? 'checked' : ''}>
```
Comment (mansaj) [inline src/views/partials/breaches.js:66]: @toufali modified here FYI

File: src/client/js/partials/breaches.js [hunk 2]
```diff
@@ -62,7 +62,7 @@ async function updateBreachStatus (input) {
       },
       body: JSON.stringify({
         affectedEmail,
-        recencyIndex: breachId,
+        breachId,
```
Comment (mansaj) [inline src/client/js/partials/breaches.js:65]: @toufali modified here FYI. For PUT we are now using breachId as the field name

**Pull request merged**